### PR TITLE
Add get store products and add products to store apis

### PR DESCRIPTION
**Pull request by dhanalakshmi-everest**

### CODE REVIEW

**Review by sammithk — COMMENTED** [review]
Please try to do the following

-  add a go.mod file
-  remove .idea file.
-  add a sql script, for creating tables.

**Comment by dhanalakshmi-everest** [comment]
> Please try to do the following
> 
> * add a go.mod file
> * remove .idea file.
> * add a sql script, for creating tables.

Completed the above changes

File: database.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE IF NOT EXISTS products
+(
+    id SERIAL,
+    name TEXT NOT NULL,
+    price NUMERIC(10,2) NOT NULL DEFAULT 0.00,
+    CONSTRAINT products_pkey PRIMARY KEY (id)
+);
+
+CREATE TABLE IF NOT EXISTS store(
```
Comment (sammithk) [inline database.sql:9]: Please add a primary key for this table.

File: main_test.go [hunk 2]
```diff
@@ -199,4 +205,56 @@ func TestDeleteProduct(t *testing.T) {
...
+		count = 1
+	}
+
+	for i := 0; i < count; i++ {
+		a.DB.Exec("INSERT INTO products(name, price) VALUES($1, $2)", "Product "+strconv.Itoa(i+1), (i+1.0)*10)
+		a.DB.Exec("INSERT INTO store(store_id, product_id, is_available) VALUES($1, $2)", storeId, 1, true)
+	}
+}
+
+func TestGetStoreProduct(t *testing.T) {
```
Comment (sammithk) [inline main_test.go:251]: in this method, please manually check for the expected result as well.

File: database.sql (hunk 1 above)
Comment (sammithk) [reply database.sql:9]: Please reconsider setting the primary key to a combination of **store_id and product_id**.

File: database.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    store_id INT,
+    product_id INT,
+    is_avialable BOOLEAN,
+    CONSTRAINT products_pkey PRIMARY KEY (store_id),
```
Comment (sammithk) [inline database.sql]: same store can have multiple products, please reconsider changing the primary key.

Comment (dhanalakshmi-everest) [reply database.sql]: Done

File: main_test.go (hunk 2 above)
Comment (dhanalakshmi-everest) [reply main_test.go:251]: Done

File: main_test.go [hunk 4]
```diff
@@ -199,4 +206,82 @@ func TestDeleteProduct(t *testing.T) {
...
+	clearTable()
+
+	storeId := 1
+	addProductsToStore(storeId, 2)
```
Comment (sammithk) [inline main_test.go:255]: Please create mock for the dao layer.

**Comment by sammithk** [comment]
Please add unit tests for dao layer as well.

**Pull request merged**